config/settings.txt
[General]
output_dir_prefix=/alf/data/focal/alpide_systemc/sim_output/pPb_20230427_short/
verbose=false

[alpide]
chip_continuous_mode=false
data_long_enable=true
dtu_delay=10
matrix_readout_speed_fast=true
minimum_busy_cycles=8
pixel_shaping_active_time_ns=5000
pixel_shaping_dead_time_ns=200
strobe_extension_enable=false

[data_output]
data_rate_interval_ns=10000
write_event_csv=true
write_event_root=true
write_vcd=false
write_vcd_clock=false

[event]
average_event_rate_ns=2000
monte_carlo_file_type=root
qed_noise_event_rate_ns=10000
qed_noise_feed_rate_ns=5000
qed_noise_input=false
qed_noise_path=config/monte_carlo_events/pp
qed_noise_rate_ns=250
random_cluster_file=false
random_cluster_filename=/software/max/alpide_systemc/config/out.root
random_cluster_generation=true
random_cluster_histogramname=hClusterSize
random_cluster_size_mean=6
random_cluster_size_stddev=1.41
random_hit_generation=false
strobe_active_length_ns=9900
strobe_inactive_length_ns=100
trigger_delay_ns=0
trigger_filter_enable=false
trigger_filter_time_ns=0

[focal]
bunch_crossing_rate_ns=25
hit_density_layer0=0.24
hit_density_layer1=0.42
hit_multiplicity_distribution_file=config/multipl_dist_focal_raw_bins.txt
innerstring=I3_I3_O3_O6
mask_grid=false
mask_grid_constant=4
monte_carlo_file_path=/alf/data/focal/alpide_systemc/input/26apr_MBpPb/
outerstring=O3_O3_O3_O6
staves_per_quadrant=33

[its]
bunch_crossing_rate_ns=25
hit_density_layer0=18.6
hit_density_layer1=12.2
hit_density_layer2=9.1
hit_density_layer3=2.8
hit_density_layer4=2.7
hit_density_layer5=2.6
hit_density_layer6=2.6
hit_multiplicity_distribution_file=config/multipl_dist_raw_bins.txt
layer0_num_staves=1
layer1_num_staves=1
layer2_num_staves=1
layer3_num_staves=1
layer4_num_staves=1
layer5_num_staves=1
layer6_num_staves=1
monte_carlo_dir_path=config/monte_carlo_events/PbPb

[pct]
beam_end_coord_x_mm=275
beam_end_coord_y_mm=20
beam_start_coord_x_mm=-5
beam_start_coord_y_mm=-5
beam_step_mm=3.0
beam_time_per_step_us=125.0
layers="0;5;10;15;20;25;30;35;40"
monte_carlo_file_path=config/monte_carlo_events/focal/pixel_event_tree_pythia_MB_r4cm.root
num_layers=2
num_staves_per_layer=12
random_beam_stddev_mm=3.0
random_particles_per_s_mean=1E9
random_particles_per_s_stddev=0.5E9
time_frame_length_ns=10000

[simulation]
n_events=38000
random_seed=0
single_chip=false
system_continuous_mode=true
system_continuous_period_ns=10000
type=focal


src/Settings/Settings.hpp
/**
 * @file   Settings.hpp
 * @author Simon Voigt Nesbo <svn@hib.no>
 * @date   November 3, 2016
 * @brief  Header file for simulation settings file handling.
 *
 *         This file has definitions for default simulation settings, which can
 *         be used as default values or for generating the settings file if it is missing.
 */


///@defgroup settings Settings
///@{
#ifndef _SETTINGS_HPP
#define _SETTINGS_HPP

#include <QtCore>
#include <QSettings>
#include <QDir>
#include <QString>

#define DEFAULT_OUTPUT_DIR_PREFIX "sim_output"

#define DEFAULT_DATA_OUTPUT_WRITE_VCD "false"
#define DEFAULT_DATA_OUTPUT_WRITE_VCD_CLOCK "false"
#define DEFAULT_DATA_OUTPUT_WRITE_EVENT_CSV "true"
#define DEFAULT_DATA_OUTPUT_DATA_RATE_INTERVAL_NS "10000"

#define DEFAULT_SIMULATION_TYPE "its"
#define DEFAULT_SIMULATION_SINGLE_CHIP "true"
#define DEFAULT_SIMULATION_N_EVENTS "10000"
#define DEFAULT_SIMULATION_SYSTEM_CONTINUOUS_MODE "false"
#define DEFAULT_SIMULATION_SYSTEM_CONTINUOUS_PERIOD_NS "5000"
#define DEFAULT_SIMULATION_RANDOM_SEED "0"

#define DEFAULT_ALPIDE_DATA_LONG_ENABLE "true"
#define DEFAULT_ALPIDE_DTU_DELAY "10"
#define DEFAULT_ALPIDE_PIXEL_SHAPING_DEAD_TIME_NS "200"
#define DEFAULT_ALPIDE_PIXEL_SHAPING_ACTIVE_TIME_NS "6000"
#define DEFAULT_ALPIDE_MATRIX_READOUT_SPEED_FAST "true"
#define DEFAULT_ALPIDE_STROBE_EXTENSION_ENABLE "false"
#define DEFAULT_ALPIDE_MINIMUM_BUSY_CYCLES "8"
#define DEFAULT_ALPIDE_CHIP_CONTINUOUS_MODE "false"

#define DEFAULT_ITS_LAYER0_NUM_STAVES "12"
#define DEFAULT_ITS_LAYER1_NUM_STAVES "16"
#define DEFAULT_ITS_LAYER2_NUM_STAVES "20"
#define DEFAULT_ITS_LAYER3_NUM_STAVES "0"
#define DEFAULT_ITS_LAYER4_NUM_STAVES "0"
#define DEFAULT_ITS_LAYER5_NUM_STAVES "0"
#define DEFAULT_ITS_LAYER6_NUM_STAVES "0"
#define DEFAULT_ITS_HIT_MULTIPLICITY_DISTRIBUTION_FILE "config/multipl_dist_raw_bins.txt"
#define DEFAULT_ITS_BUNCH_CROSSING_RATE_NS "25"
#define DEFAULT_ITS_MONTE_CARLO_DIR_PATH "config/monte_carlo_events/PbPb"
#define DEFAULT_ITS_HIT_DENSITY_LAYER0 "18.6"
#define DEFAULT_ITS_HIT_DENSITY_LAYER1 "12.2"
#define DEFAULT_ITS_HIT_DENSITY_LAYER2 "9.1"
#define DEFAULT_ITS_HIT_DENSITY_LAYER3 "2.8"
#define DEFAULT_ITS_HIT_DENSITY_LAYER4 "2.7"
#define DEFAULT_ITS_HIT_DENSITY_LAYER5 "2.6"
#define DEFAULT_ITS_HIT_DENSITY_LAYER6 "2.6"

#define DEFAULT_PCT_LAYERS "0;5;10;15;20;25;30;35;40"
#define DEFAULT_PCT_NUM_STAVES_PER_LAYER "1"
#define DEFAULT_PCT_MONTE_CARLO_FILE_PATH "config/monte_carlo_events/pct/pct_data.xml"
#define DEFAULT_PCT_TIME_FRAME_LENGTH_NS "200"
#define DEFAULT_PCT_RANDOM_PARTICLES_PER_S_MEAN "1E9"
#define DEFAULT_PCT_RANDOM_PARTICLES_PER_S_STDDEV "0.2E9"
#define DEFAULT_PCT_RANDOM_BEAM_STDDEV_MM "0.1"
#define DEFAULT_PCT_BEAM_START_COORD_X_MM "-0.5"
#define DEFAULT_PCT_BEAM_START_COORD_Y_MM "-0.5"
#define DEFAULT_PCT_BEAM_END_COORD_X_MM "27.5"
#define DEFAULT_PCT_BEAM_END_COORD_Y_MM "2.0"
#define DEFAULT_PCT_BEAM_STEP_MM "3.0"
#define DEFAULT_PCT_BEAM_TIME_PER_STEP_US "125"

#define DEFAULT_FOCAL_MONTE_CARLO_FILE_PATH "config/monte_carlo_events/focal/pixel_event_tree_pythia_MB_r4cm.root"
#define DEFAULT_FOCAL_STAVES_PER_QUADRANT "3"
#define DEFAULT_FOCAL_INNER_STRING "I3_I3_O3_O6"
#define DEFAULT_FOCAL_OUTER_STRING "O3_O3_O3_O3_O3"
#define DEFAULT_FOCAL_BUNCH_CROSSING_RATE_NS "25"
#define DEFAULT_FOCAL_HIT_MULTIPLICITY_DISTRIBUTION_FILE "config/multipl_dist_focal_raw_bins.txt"
#define DEFAULT_FOCAL_HIT_DENSITY_LAYER0 "0.1"
#define DEFAULT_FOCAL_HIT_DENSITY_LAYER1 "0.1"

#define DEFAULT_EVENT_RANDOM_HIT_GENERATION "true"
#define DEFAULT_EVENT_RANDOM_CLUSTER_GENERATION "false"
#define DEFAULT_EVENT_RANDOM_CLUSTER_SIZE_MEAN "4"
#define DEFAULT_EVENT_RANDOM_CLUSTER_SIZE_STDDEV "2"
#define DEFAULT_EVENT_MONTE_CARLO_FILE_TYPE "xml"
#define DEFAULT_EVENT_QED_NOISE_PATH "config/monte_carlo_events/QED"
#define DEFAULT_EVENT_QED_NOISE_INPUT "false"
#define DEFAULT_EVENT_QED_NOISE_FEED_RATE_NS "250"
#define DEFAULT_EVENT_QED_NOISE_EVENT_RATE_NS "20000"
#define DEFAULT_EVENT_TRIGGER_DELAY_NS "1000"
#define DEFAULT_EVENT_TRIGGER_FILTER_TIME_NS "10000"
#define DEFAULT_EVENT_TRIGGER_FILTER_ENABLE "true"
#define DEFAULT_EVENT_STROBE_ACTIVE_LENGTH_NS "100"
#define DEFAULT_EVENT_STROBE_INACTIVE_LENGTH_NS "100"
#define DEFAULT_EVENT_AVERAGE_EVENT_RATE_NS "2500"

QSettings *getSimSettings(const char *fileName = "config/settings.txt");
void setDefaultSimSettings(QSettings *readoutSimSettings);


#endif
///@}
